src/sources/soundsourceflac.cpp

```diff
@@ -193,8 +199,12 @@ ReadableSampleFrames SoundSourceFLAC::readSampleFramesClamped(
                 && (precedingFrames != readSampleFramesClamped(
                         WritableSampleFrames(precedingFrames)).frameIndexRange())) {
             kLogger.warning()
-                    << "Failed to skip preceding frames"
+                    << "Resetting decoder after failure to skip preceding frames"
                     << precedingFrames;
+            if (!FLAC__stream_decoder_reset(m_decoder)) {
+                kLogger.critical()
```

Comment:
This crashes Mixxx, right? Doe we have alternatives?

Reply:
Not by default: "It exits if the environment variable QT_FATAL_CRITICALS is not empty."

Reply:
If even resetting the decoder fails then we are in big trouble. Like a destructor this operation should never fail, but the API allows failure and we need to handle this case.

Reply:
Playback was stuttering and Mixxx became almost unusable when not resetting the decoder. This is already our very last chance. If even this option fails everything is lost.

This is still only a DJ application, not the control unit of some deadly weapon. Let's keep things in perspective.